packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/RealmSyncUtils.kt

```diff
@@ -52,208 +52,226 @@ internal fun convertSyncErrorCode(error: SyncErrorCode): SyncException {
     // FIXME Client Reset errors are just reported as normal Sync Errors for now.
     //  Will be fixed by https://github.com/realm/realm-kotlin/issues/417
     val message = createMessageFromSyncError(error)
-    return when (error.category) {
-        SyncErrorCodeCategory.RLM_SYNC_ERROR_CATEGORY_CLIENT -> {
-            // See https://github.com/realm/realm-core/blob/master/src/realm/sync/client_base.hpp#L73
-            // For now, it is unclear how to categorize these, so for now, just report as generic
-            // errors.
-            SyncException(message)
-        }
-        SyncErrorCodeCategory.RLM_SYNC_ERROR_CATEGORY_CONNECTION -> {
-            // See https://github.com/realm/realm-core/blob/master/src/realm/sync/protocol.hpp#L200
-            // Use https://docs.google.com/spreadsheets/d/1SmiRxhFpD1XojqCKC-xAjjV-LKa9azeeWHg-zgr07lE/edit
-            // as guide for how to categorize Connection type errors.
-            when (ProtocolConnectionErrorCode.fromInt(error.value)) {
-                ProtocolConnectionErrorCode.RLM_SYNC_ERR_CONNECTION_UNKNOWN_MESSAGE, // Unknown type of input message
-                ProtocolConnectionErrorCode.RLM_SYNC_ERR_CONNECTION_BAD_SYNTAX, // Bad syntax in input message head
-                ProtocolConnectionErrorCode.RLM_SYNC_ERR_CONNECTION_WRONG_PROTOCOL_VERSION, // Wrong protocol version (CLIENT) (obsolete)
-                ProtocolConnectionErrorCode.RLM_SYNC_ERR_CONNECTION_BAD_SESSION_IDENT, // Bad session identifier in input message
-                ProtocolConnectionErrorCode.RLM_SYNC_ERR_CONNECTION_REUSE_OF_SESSION_IDENT, // Overlapping reuse of session identifier (BIND)
-                ProtocolConnectionErrorCode.RLM_SYNC_ERR_CONNECTION_BOUND_IN_OTHER_SESSION, // Client file bound in other session (IDENT)
-                ProtocolConnectionErrorCode.RLM_SYNC_ERR_CONNECTION_BAD_MESSAGE_ORDER, // Bad input message order
-                ProtocolConnectionErrorCode.RLM_SYNC_ERR_CONNECTION_BAD_DECOMPRESSION, // Error in decompression (UPLOAD)
-                ProtocolConnectionErrorCode.RLM_SYNC_ERR_CONNECTION_BAD_CHANGESET_HEADER_SYNTAX, // Bad syntax in a changeset header (UPLOAD)
-                ProtocolConnectionErrorCode.RLM_SYNC_ERR_CONNECTION_BAD_CHANGESET_SIZE -> { // Bad size specified in changeset header (UPLOAD)
-                    UnrecoverableSyncException(message)
-                }
-                ProtocolConnectionErrorCode.RLM_SYNC_ERR_CONNECTION_SWITCH_TO_FLX_SYNC, // Connected with wrong wire protocol - should switch to FLX sync
-                ProtocolConnectionErrorCode.RLM_SYNC_ERR_CONNECTION_SWITCH_TO_PBS -> { // Connected with wrong wire protocol - should switch to PBS
-                    WrongSyncTypeException(message)
+    return safeErrorCodeMapping(onException = { SyncException(it) }) {
+        when (error.category) {
```

Comment:
Probably not worth fixing in this PR, but we would still crash if we suddenly see a category we haven't seen before. It would crash here though, but inside RealmInterop. Maybe just create an issue for it for now?